blocks/components/editable/format-toolbar.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+ // TODO: We mustn't import by relative path traversing from blocks to editor
+ // as we're doing here; instead, we should consider a common components path.
+import IconButton from '../../../editor/components/icon-button';
+
+const FORMATTING_CONTROLS = [
+	{
+		icon: 'editor-bold',
+		title: wp.i18n.__( 'Bold' ),
+		format: 'bold'
+	},
+	{
+		icon: 'editor-italic',
+		title: wp.i18n.__( 'Italic' ),
+		format: 'italic'
+	},
+	{
+		icon: 'editor-strikethrough',
+		title: wp.i18n.__( 'Strikethrough' ),
+		format: 'strikethrough'
+	}
+];
+
+class FormatToolbar extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			linkValue: ''
+		};
+		this.addLink = this.addLink.bind( this );
+		this.dropLink = this.dropLink.bind( this );
+		this.submitLink = this.submitLink.bind( this );
+		this.updateLinkValue = this.updateLinkValue.bind( this );
+	}
+
+	componentWillMount() {
+		this.setState( { linkValue: this.props.formats.link } );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.setState( { linkValue: nextProps.formats.link } );
+	}
+
+	toggleFormat( format ) {
+		return ( event ) => {
+			event.stopPropagation();
+			this.props.onChange( {
+				[ format ]: ! this.props.formats[ format ]
+			} );
+		};
+	}
+
+	addLink() {
+		if ( ! this.props.formats.link ) {
+			// TODO find a way to add an empty link to TinyMCE
+			this.props.onChange( { link: 'http://wordpress.org' } );
```

Comment:
In core we put `_wp_link_placeholder` as the `href` and then make sure it's stripped everywhere appropriate. :)